modules/vault-cluster/main.tf

```diff
@@ -8,6 +8,41 @@ terraform {
   required_version = ">= 0.10.3"
 }
 
+# ---------------------------------------------------------------------------------------------------------------------
+# CREATES A SERVICE ACCOUNT TO OPERATE THE VAULT CLUSTER
+# The default project service account will be used if create_service_account
+# is set to false and no service_account_email is provided.
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "google_service_account" "vault_cluster_admin" {
+  count        = "${var.create_service_account}"
+  account_id   = "${var.cluster_name}-admin-sa"
+  display_name = "Vault Server Admin"
+  project      = "${var.gcp_project_id}"
+}
+
+# Create a service account key
+resource "google_service_account_key" "vault" {
+  count              = "${var.create_service_account}"
+  service_account_id = "${google_service_account.vault_cluster_admin.name}"
+}
+
+# Add viewer role to service account on project
+resource "google_project_iam_member" "vault_cluster_admin_sa_view_project" {
+  count   = "${var.create_service_account}"
+  project = "${var.gcp_project_id}"
+  role    = "roles/viewer"
+  member  = "serviceAccount:${google_service_account.vault_cluster_admin.email}"
```

Comment:
So is this adding a single _user_ as an admin? Why do we need that for this module? Don't we only need to add permissions for a compute instance, not a person?

Reply:
The email in this case is not `someone@gruntwork.io` but auto generated by google and I think they're used only for identification purposes. 

For example `project_id@appspot.gserviceaccount.com`, it depends on the type of service account.

Reply:
Ohhh, I gotcha. Thanks for the explanation.